format-common/src/test/java/io/cdap/plugin/format/input/PathTrackingConfigTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright ©2022 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.plugin.format.input;
+
+import io.cdap.plugin.common.batch.JobUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Job;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class PathTrackingConfigTest {
+  String pathStr;
+  Job job;
+  Configuration hconf;
+  List<Path> expectedPaths;
```

Comment:
Avoid using field for this as it can get accumulated across tests. Use local variable instead.

Reply:
Got it :)